packages/reports/tests/unit/consumers/request/logical-table-test.js

```diff
@@ -2,47 +2,60 @@ import { module, test } from 'qunit';
 import { setupTest } from 'ember-qunit';
 import { RequestActions } from 'navi-reports/services/request-action-dispatcher';
 
-const DAY = { name: 'day' },
-  WEEK = { name: 'week' };
-
 module('Unit | Consumer | request logical table', function(hooks) {
   setupTest(hooks);
 
   test('UPDATE_TABLE', function(assert) {
-    assert.expect(5);
+    assert.expect(9);
 
-    const MockDispatcher = {
-      dispatch(action, route, timeGrain) {
-        assert.equal(action, RequestActions.ADD_TIME_GRAIN, 'ADD_TIME_GRAIN is sent as part of UPDATE_TABLE');
+    const logicalTable = { table: 'oldTable', timeGrain: 'day' };
+    const currentModel = { request: { logicalTable, dataSource: 'oldDataSource' } };
+    const newTable = { name: 'newTable', timeGrains: ['day', 'week'], source: 'newDataSource' };
 
-        assert.equal(
-          timeGrain,
-          DAY,
-          'When the old and new tables share a time grain, that grain is given to ADD_TIME_GRAIN'
-        );
+    let expectTableAction = false;
+
+    const MockDispatcher = {
```

Comment:
Maybe instead of a boolean for which action, add a `timesCalled = 0` property and then if/else on that. I'm ok with not changing it but it makes it a little less straightforward to read